From f4c7f897a440afd18bf979498d06419b3a8bf116 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:59 -0400
Subject: [ethernet] qlcnic: Use bool for rx_mac_learn

Message-id: <1408626422-3941-76-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88657
O-Subject: [PATCH RHEL 7.1 075/138] qlcnic: Use bool for rx_mac_learn.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 72ebe3495f6f6829306ccf6c6a8b1b92c2c9b672

o Use boolean type instead of u8.

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: Himanshu Madhani <himanshu.madhani@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
index e154781..7ad6348 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
@@ -1082,7 +1082,7 @@ struct qlcnic_adapter {
  u64 dev_rst_time;
  bool drv_mac_learn;
  bool fdb_mac_learn;
- u8 rx_mac_learn;
+ bool rx_mac_learn;
  unsigned long vlans[BITS_TO_LONGS(VLAN_N_VID)];
  u8 flash_mfg_id;
  struct qlcnic_npar_info *npars;
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index b3f9762..27cd454 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -1637,7 +1637,7 @@ static void qlcnic_83xx_set_interface_id_promisc(struct qlcnic_adapter *adapter,
  if (qlcnic_sriov_pf_check(adapter)) {
   qlcnic_alloc_lb_filters_mem(adapter);
   qlcnic_pf_set_interface_id_promisc(adapter, interface_id);
-  adapter->rx_mac_learn = 1;
+  adapter->rx_mac_learn = true;
  } else {
   if (!qlcnic_sriov_vf_check(adapter))
    *interface_id = adapter->recv_ctx->context_id << 16;
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
index 6d7f6a1..f19cec8 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
@@ -2217,7 +2217,7 @@ int qlcnic_83xx_init(struct qlcnic_adapter *adapter, int pci_using_dac)
  struct qlcnic_hardware_context *ahw = adapter->ahw;
  int err = 0;
 
- adapter->rx_mac_learn = 0;
+ adapter->rx_mac_learn = false;
  ahw->msix_supported = !!qlcnic_use_msi_x;
 
  qlcnic_83xx_init_rings(adapter);
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c
index e26c49f..be7d7a6 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_vnic.c
@@ -227,10 +227,10 @@ int qlcnic_83xx_config_vnic_opmode(struct qlcnic_adapter *adapter)
  if (ahw->capabilities & QLC_83XX_ESWITCH_CAPABILITY) {
   adapter->flags |= QLCNIC_ESWITCH_ENABLED;
   if (adapter->drv_mac_learn)
-   adapter->rx_mac_learn = 1;
+   adapter->rx_mac_learn = true;
  } else {
   adapter->flags &= ~QLCNIC_ESWITCH_ENABLED;
-  adapter->rx_mac_learn = 0;
+  adapter->rx_mac_learn = false;
  }
 
  ahw->idc.vnic_state = QLCNIC_DEV_NPAR_NON_OPER;
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
index c549a43..2cd211c 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
@@ -547,10 +547,10 @@ static void __qlcnic_set_multi(struct net_device *netdev, u16 vlan)
   qlcnic_alloc_lb_filters_mem(adapter);
   adapter->drv_mac_learn = 1;
   if (adapter->flags & QLCNIC_ESWITCH_ENABLED)
-   adapter->rx_mac_learn = 1;
+   adapter->rx_mac_learn = true;
  } else {
   adapter->drv_mac_learn = 0;
-  adapter->rx_mac_learn = 0;
+  adapter->rx_mac_learn = false;
  }
 
  qlcnic_nic_set_promisc(adapter, mode);
-- 
1.7.1